### PR TITLE
Adding new pandas example

### PR DESCRIPTION
**Pull request by atravitz**
There was a previous pandas example notebook but it used the old api that required manually creating a dataframe from a signac index. We now have `to_dataframe()` and I'm rewriting the example to make use of this.

### CODE REVIEW

File: notebooks/signac_202_Integration_with_pandas.ipynb
Comment (bdice) [inline notebooks/signac_202_Integration_with_pandas.ipynb:118]:
```suggestion
    "Now we can select specific data subsets, for example simulations with pressure greater than 0.1 :"
```

**Comment by syjlee** [comment]
@atravitz I think it's a great simple example to show how pandas can be used with signac together. But when I run the notebook files up to 202, I get the keyerror that doc.V doesn't exist. From running previous notebooks before 202, I have doc.V_liq and doc.V_gas, not doc.V. Did I miss something somewhere?

**Review by csadorf — COMMENTED** [review]
I read the the NB-review, but did not run the code. I am wondering whether we should use [pandas plot function](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.plot.html) (see also [pandas visualization](https://pandas.pydata.org/pandas-docs/stable/user_guide/visualization.html))  for the last cell, otherwise LGTM!

**Comment by cbkerr** [comment]
> @atravitz I think it's a great simple example to show how pandas can be used with signac together. But when I run the notebook files up to 202, I get the keyerror that doc.V doesn't exist. From running previous notebooks before 202, I have doc.V_liq and doc.V_gas, not doc.V. Did I miss something somewhere?

I just saw that the same thing was caught by the integration test. I just ran it successfully on notebooks 103 and 202. `V_liq` and `V_gas` could have been from an older example

**Review by cbkerr — APPROVED** [review]
I think we should go ahead and merge this because I found it helpful when trying to figure out how to analyze a growing data space.

**Review by syjlee — APPROVED** [review]

**Comment by bdice** [comment]
I am currently addressing issues with the "projects" CI in a separate branch and noticed that the "notebooks" CI is failing for the pandas example on that branch. Therefore, I am working on finishing this PR and fixing the notebooks CI. I am pushing changes to address the following issues. I will iterate until CI passes, then merge this PR.

- CI is failing. This should be a red flag for reviewers. I recommend that _reviewers should not approve PRs with failures in CI_, except in specific cases where the CI failure is not caused by the contents of the pull request (and is being addressed in a separate PR or issue). We have CI robots to help humans identify problems in the code, and should be cautious about merging code that has faults.
- This PR indicates that notebook 103 should be run first, but the committed contents of this notebook do not reflect that state. The project that is created in notebook 101 has 3 state points, and is expanded in 103 to include a total of 14 state points. However, the committed notebooks only show 3 state points.
- The missing state points are concealing a bug in the code. When used with only 3 state points, the state points' ids happen to be alphabetized in a way that orders them according to pressure. However, if there are 14 state points, we must add a line `df = df.sort_values("sp.p")` to account for the sorting, or else the plot line is drawn in a random order.
- Notebook 103 should indicate a dependence on notebook 101, just like notebook 202 indicates its dependence on notebook 103. This is an unrelated improvement to the core improvements to notebook 202, but is probably important for users.

**Comment by bdice** [comment]
The CI issue with not finding `doc.V` is because of notebook 104 -- the van der Waals approximations with parameters `a` and `b` updates the original state points from notebook 103 to include `a=0` and `b=0`, and relabels `doc.V` to `doc.V_gas`.

I spent some time trying to untangle the data space modifications in notebooks 103 and 104, but it seems to me that the better solution would be to make this "advanced" notebook self-contained and eliminate the dependency on notebooks 103/104. The fastest way to do that is to create a temporary project and plot a function. I'm going to make that change, while retaining the overall style of this PR and try to keep it as simple as possible.

**Pull request merged**
